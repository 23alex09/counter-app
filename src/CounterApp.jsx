import { useState } from 'react'
import PropTypes from 'prop-types';

export const CounterApp = ( { value } ) => {

    const [ counter, setCounter ] = useState( value );

    const handleAdd = () =>
        setCounter( counter + 1 );


    const handleSubstract = () =>
        setCounter( counter - 1 );


    const handleReset = () => setCounter( 0 );


    return (
        <>
            <h1>Counter App</h1>
            <h2>{ counter }</h2>
            <button aria-label="btn-inc" onClick={ handleAdd }> +1 </button>
            <button aria-label="btn-dec" onClick={ handleSubstract }> -1 </button>
            <button aria-label="btn-reset" onClick={ handleReset }> Reset </button>
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}
