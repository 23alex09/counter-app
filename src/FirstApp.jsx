import PropTypes from 'prop-types';

export const FirstApp = ( { title, subtitle, name } ) => {
    //console.log(props)
    return (
        <>
            <h1 data-testid="test-title">{ title }</h1>
            <h2>{ subtitle }</h2>
            <p>{ name }</p>
        </>
    )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.number
}

FirstApp.defaultProps = {
    subtitle: 1000,
    name: 'Alejandro'
}