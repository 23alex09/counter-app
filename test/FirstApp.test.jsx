import { render, screen } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"

describe( 'test suit for FirstApp component', () => {

    const title = 'Hello, World!';
    const subtitle = 123;

    test( 'should match with the snapshot', () => {
        const { container } = render( <FirstApp title={ title } /> )
        expect( container ).toMatchSnapshot();
    } )


    test( 'should show the title Hello, World!', () => {
        render( <FirstApp title={ title } /> );
        expect( screen.getByText( title ) ).toBeTruthy();
    } )

    test( 'should show the title within a h1', () => {

        render( <FirstApp title={ title } /> );
        expect( screen.getByRole( 'heading', { level: 1 } ).innerHTML ).toContain( title );

    } )

    test( 'should show the subtitle send by props', () => {

        render( <FirstApp title={ title } subtitle={ subtitle } /> )
        expect( screen.getByText( subtitle ) ).toBeTruthy();

    } )
} )