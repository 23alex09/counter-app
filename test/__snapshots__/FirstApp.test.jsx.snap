// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test suit for FirstApp component should match with the snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hello, World!
  </h1>
  <h2>
    1000
  </h2>
  <p>
    Alejandro
  </p>
</div>
`;
